SimpleNotePad.java

```diff
@@ -0,0 +1,225 @@
+import java.awt.Dimension;
+import java.awt.Graphics;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.print.PageFormat;
+import java.awt.print.Printable;
+import java.awt.print.PrinterException;
+import java.awt.print.PrinterJob;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.stream.Stream;
+
+import javax.swing.JFileChooser;
+import javax.swing.JFrame;
+import javax.swing.JMenu;
+import javax.swing.JMenuBar;
+import javax.swing.JMenuItem;
+import javax.swing.JOptionPane;
+import javax.swing.JScrollPane;
+import javax.swing.JTextPane;
+
+/*
+ * Code Smell Refactoring:
+ * -Changed the variables to private instead of public
+ * -Changed variable names at the beginning of class to make it more readable
+ * -Line 145: Removed undo feature since it is not in the design specs.
+ * -Split the file and edit menu items to separate methods to make it more
+ * 	readable.
+ * -Removed the println in the paste block. 
+ * -Moved some of the code from the if-else block in the actionPerformed method
+ *  to separate methods to make it more readable.
+ * -Removed unused code from paste function
+ * 
+ * -Added new features in the assignment spec
+ * 
+ */
+
+public class SimpleNotePad extends JFrame implements ActionListener{
+	private JMenuBar menuBar;
+	private JMenu fileMenu, editMenu, openRecentFile;
+	private JTextPane d = new JTextPane();
+	private JMenuItem newFile, saveFile, printFile, openFile;
+	private JMenuItem copy, paste, simpleReplace;
+	private JMenuItem[] recent = new JMenuItem[5];
+	private File[] recentFile = {null, null, null, null, null};
+	
+    public SimpleNotePad() {
+        setTitle("A Simple Notepad Tool");
+        menuBar = new JMenuBar();
+        initFileMenu();
+        initEditMenu();
+        menuBar.add(fileMenu);
+        menuBar.add(editMenu);
+        setJMenuBar(menuBar);
+        add(new JScrollPane(d));
+        setPreferredSize(new Dimension(600,600));
+        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        setVisible(true);
+        pack();
+    }
+    
+    private void initFileMenu() {
+    	fileMenu = new JMenu("File");
+    	openRecentFile = new JMenu("Open Recent File");
+    	
+    	openFile = new JMenuItem("Open File");
+    	newFile = new JMenuItem("New File");
+    	saveFile = new JMenuItem("Save File");
+    	printFile = new JMenuItem("Print File");
+    	
+    	fileMenu.add(openFile);
+    	fileMenu.addSeparator();
+    	fileMenu.add(newFile);
+        fileMenu.addSeparator();
+        fileMenu.add(saveFile);
+        fileMenu.addSeparator();
+        fileMenu.add(printFile);
+        fileMenu.addSeparator();
+        fileMenu.add(openRecentFile);
+    	
+        openFile.addActionListener(this);

```

Comment:
```suggestion
        openFile.addActionListener(e -> {
            // call the desired function here
        });
```
ActionListner is a functional interface, so you can pass in a lambda expression to addActionListener. This will create an anonymous object the implements ActionListner, and the body of the expression becomes the body of the actionPerformed method of ActionListner.

I recommend you make this change to all addActionListner statements and remove all setActionCommand statements throughout the file. Then you no longer need work directly actionPerformed at all, so SimpleNotePad does not need to implement ActionListener, and the imports for ActionListener and ActionEvent are no longer needed as well. Additionally, because actionPerformed does not need to be overridden, all of those if/else-if blocks go away entirely, which benefits the readability of your code as well as a slight improvement to the speed of all the menu operations.